bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -66,7 +66,9 @@ def run_plan(scope, plan_name, named_args = {})
         # If the plan does not throw :return by calling the return function it's result is
         # undef/nil
         result = catch(:return) do
-          func.class.dispatcher.dispatchers[0].call_by_name_with_scope(scope, params, true)
+          scope.with_global_scope do |global_scope|
```

Comment:
can you move this commit to a separate PR against master and reference https://tickets.puppetlabs.com/browse/BOLT-1190